nasl/nasl_packet_forgery.c

```diff
@@ -1437,7 +1438,16 @@ nasl_send_packet (lex_ctxt *lexic)
       bzero (&sockaddr, sizeof (struct sockaddr_in));
       sockaddr.sin_family = AF_INET;
       sockaddr.sin_addr = sip->ip_dst;
-      if (sockaddr.sin_addr.s_addr != inaddr.s_addr)
+
+      if (allow_broadcast)
+        {
+          if (setsockopt (soc, SOL_SOCKET, SO_BROADCAST, &i, sizeof (i)) < 0)
+            perror ("setsockopt ");
+          if (sockaddr.sin_addr.s_addr != INADDR_BROADCAST)
+            allow_broadcast = 0;
```

Comment:
I don't know the context very well, so I am wondering why you would set the socket to broadcast above, if you're going to disable broadcasting here.

Reply:
allow_broadcast will be disabled if the allow broadcast flag was enabled but the destination address differs from the broadcast address. 
In the next code line, it will check if the the dst ip differs from the target ip. In that case it will return immediately, except that the dst ip is the broadcast address and the allow_broadcast is set to 1.

Reply:
OK thanks, I think I get it.